### PR TITLE
chore(button): CSS for icon-only attribute

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description
Added CSS for icon-only attribute for `sp-button`

## Motivation and context

To solve the below issue on a rounded button with icon
Screenshot 2023-10-11 at 7 01 14 PM

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ ] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
FYI: https://github.com/adobe/spectrum-web-components/issues/3398

**Review by najikahalsema — APPROVED** [review]

**Comment by Rajdeepc** [comment]
@najikahalsema @Westbrook If everything looks good here can we merge this?

**Review by najikahalsema — APPROVED** [review]

**Review by najikahalsema — DISMISSED** [review]
Disregard my approval. Can you investigate why the build is failing to meet test coverage in chromium?

**Comment by Rajdeepc** [comment]
> Disregard my approval. Can you investigate why the build is failing to meet test coverage in chromium?

The build is failing due to code coverage. Can we take this up in a separate ticket as this is unrelated to the changes in this PR?
Screenshot 2023-11-02 at 12 15 49 AM

**Comment by Rajdeepc** [comment]
> Disregard my approval. Can you investigate why the build is failing to meet test coverage in chromium?

Should be good now

**Review by Westbrook — COMMENTED** [review]
Some things for this:
- please add stories with the icon only button in isolation so we can manage it via regression testing
- Should a user be able to set `buttonRef.iconOnly = true` or do we want to require `buttonRef.toggleAttribute('icon-only', true)`? Either way works, but we should add content about this to the documentation site based on the decisions we make there for today.
- see https://spectrum.adobe.com/page/button/#Label-and-icon, particularly:
  > The label can be hidden to create an icon-only button. If the label is hidden, an icon is required, and the label will appear in a tooltip.
  
  I'm OK not to block on that, though it would be nice to have this feature, too. We can land this for Coachmark work as is (with stories) and then keep #3398 open for the follow up work around a Tooltip.

**Comment by Rajdeepc** [comment]
> Some things for this:
> 
> * please add stories with the icon only button in isolation so we can manage it via regression testing
> * Should a user be able to set `buttonRef.iconOnly = true` or do we want to require `buttonRef.toggleAttribute('icon-only', true)`? Either way works, but we should add content about this to the documentation site based on the decisions we make there for today.
> * see https://spectrum.adobe.com/page/button/#Label-and-icon, particularly:
>   > The label can be hidden to create an icon-only button. If the label is hidden, an icon is required, and the label will appear in a tooltip.
>   
>   
>   I'm OK not to block on that, though it would be nice to have this feature, too. We can land this for Coachmark work as is (with stories) and then keep [[Feat][Button]: Support `icon-only` attribute #3398](https://github.com/adobe/spectrum-web-components/issues/3398) open for the follow up work around a Tooltip.

@Westbrook Added the required stories and also updated the tests and docs to cater icon-only attribute for Icon Only Button element

File: packages/button/test/button.test.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ describe('Button', () => {
     it('loads default only icon', async () => {
         const el = await fixture<Button>(
             html`
-                <sp-button label="Button">
+                <sp-button label="Button" icon-only>
```
Comment (Westbrook) [inline packages/button/test/button.test.ts:87]: 👏🏼 👏🏼 👏🏼

**Review by Westbrook — APPROVED** [review]
LGTM. Let's land this plane 🛬

**Pull request merged**